xcmailr-webapp/src/main/java/controllers/AdminHandler.java

```diff
@@ -380,29 +379,30 @@ else if (action.equals("deleteDomain"))
      * @return overview of all white-listed domains
      */
     @FilterWith(WhitelistFilter.class)
-    public Result addDomain(Context context, @Param("domainName") String domainName)
+    public Result addDomain(Context context, @Param("domainName") Optional<String> domainName)
```

Comment:
Is `domainName` really optional? The UI always sends a value.

Reply:
This is to allow custom error handling in case domainName is null or blank.

Reply:
Reverted to non-optional parameter.